compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateAIEWorkgroup.cpp

```diff
@@ -70,24 +71,56 @@ LogicalResult workgroupBuildForDmaCpyNdOp(
     Block::iterator controlCodeBegin, Block::iterator controlCodeEnd) {
   LLVM_DEBUG(llvm::dbgs() << "workgroupBuild [amdaie.dma_cpy_nd] Start\n");
   Attribute sourceMemSpace = dmaOp.getSourceObjectFifo().getMemorySpace();
+  Attribute targetMemSpace = dmaOp.getTargetObjectFifo().getMemorySpace();
   Location loc = rewriter.getUnknownLoc();
   SmallVector<OpFoldResult> empty;
+  SmallVector<Value> emptyValue =
+      getValueOrCreateConstantIndexOp(rewriter, loc, empty);
+  SmallVector<SmallVector<Value>> circularTargetData(3, emptyValue);
```

Comment:
I think vector `vector<vector<Value>>` makes is unclear what the indices refer to and using separate variables with clear names would improve clarity and won't be (much) more verbose than using vectors. Also, I think you can directly use `OpFoldResult` instead of `Value` to avoid `getValueOrCreateConstantIndexOp` and `toOpFoldResults`:

```
SmallVector<OpFoldResult> circularDmaTargetOffsets;
SmallVector<OpFoldResult> circularDmaTargetSizes;
SmallVector<OpFoldResult> circularDmaTargetStrides;
SmallVector<OpFoldResult> circularDmaSourceOffsets;
SmallVector<OpFoldResult> circularDmaSourceSizes;
SmallVector<OpFoldResult> circularDmaSourceStrides;
```

Reply:
This was really helpful. Wish I saw this comment before trying to fix the lit tests. :P